test/async-emitter.js

```diff
@@ -0,0 +1,214 @@
+'use strict';
+
+const { AsyncEmitter } = require('..');
+const metatests = require('metatests');
+
+metatests.test('AsyncEmitter on/emit', async test => {
+  const ae = new AsyncEmitter();
+  let emitted = false;
+
+  ae.on('e1', async (a, b, c, d) => {
+    test.strictSame(a, 1);
+    test.strictSame(b, 2);
+    test.strictSame(c, 3);
+    test.strictSame(d, 4);
+    emitted = true;
+  });
+
+  await ae.emit('e1', 1, 2, 3, 4);
+
+  test.strictSame(emitted, true);
+  test.end();
```

Comment:
These 2 are not needed, metatests will wait on the promise returned from the test if any and call `end` when it resolves.